### PR TITLE
upgraded evictions_number metric to stable

### LINKED ISSUE #105864: Proposed graduation of evictions_number metric to stable
### What would you like to be added?

I would like to propose the graduation of the [evictions_number](https://github.com/kubernetes/kubernetes/blame/0d50c969c587c8a6c16e0962118305ac652c5a6b/pkg/controller/nodelifecycle/metrics.go#L67) metric from alpha to stable.

### Why is this needed?

The `evictions_number` metric has been stable for 2 years now.

### PR DESCRIPTION
**Pull request by cyclinder**
Signed-off-by: cyclinder <qifeng.guo@daocloud.io>

#### What type of PR is this?
/kind feature
/kind cleanup

#### What this PR does / why we need it:
upgraded evictions_number metric to stable
#### Which issue(s) this PR fixes:

Fixes #105864

#### Does this PR introduce a user-facing change?

```release-note
Rename metrics `evictions_number` to `evictions_total` and mark it as stable. The original `evictions_number` metrics name is marked as "Deprecated" and will be removed in kubernetes 1.23.0. 

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by dgrisonnet — COMMENTED** [review]
The changes look good to me overall, you just need to format your code to match what gofmt expects: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105935/pull-kubernetes-verify/1453654163291377664

**Review by dgrisonnet — COMMENTED** [review]
Could please add a release note to your PR to mention that the `node_collector_evictions_number` metric is replaced by `node_collector_evictions_total` and that the latter will be graduated to stable?

**Comment by cyclinder** [comment]
@aveshagarwal @logicalhan PTAL

File: pkg/controller/nodelifecycle/metrics.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ const (
 	zoneSizeKey             = "zone_size"
 	zoneNoUnhealthyNodesKey = "unhealthy_nodes_in_zone"
 	evictionsNumberKey      = "evictions_number"
+	evictionsTotalKey       = "evictions_total"
```
Comment (logicalhan) [inline pkg/controller/nodelifecycle/metrics.go:32]: can you inline this? it's only used once so it's confusing when i can't see it in directly in the metric

Comment (cyclinder) [reply pkg/controller/nodelifecycle/metrics.go:32]: thank your review! I don't the mean of inline , Can you explain or tell me what to do, thanks

Comment (jan--f) [reply pkg/controller/nodelifecycle/metrics.go:32]: Wouldn't it make sense to keep it as is for consistency's sake? The other `*Key` constants are only used once as well.

**Review by jan--f — APPROVED** [review]
/lgtm

File: pkg/controller/nodelifecycle/metrics.go (hunk 1 above)
Comment (cyclinder) [reply pkg/controller/nodelifecycle/metrics.go:32]: @logicalhan Does it need to be modified here?

Comment (dgrisonnet) [reply pkg/controller/nodelifecycle/metrics.go:32]: For consistency purposes, I think it is fine to keep it there, although I agree with @logicalhan that it is confusing to have it defined there.

/lgtm

Comment (cyclinder) [reply pkg/controller/nodelifecycle/metrics.go:32]: Thanks, Done

**Comment by cyclinder** [comment]
@dashpole @logicalhan PTAL

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Comment by cyclinder** [comment]
@bowei  @k82cn  PTAL

**Pull request closed without merging**